src/Uno.UI/UI/Xaml/Application.macOS.cs

```diff
@@ -154,11 +171,15 @@ private void SetCurrentLanguage()
 		}
 
 		partial void ObserveSystemThemeChanges()
-		{			
-			NSUserDefaults.StandardUserDefaults.AddObserver(
-				"AppleInterfaceStyle",
-				NSKeyValueObservingOptions.New,
-				_ => Application.Current.OnSystemThemeChanged());
+		{
+			NSDistributedNotificationCenter.GetDefaultCenter().AddObserver(
```

Comment:
Was that related to a race condition ? or just based on OS version?

Reply:
Apparently this is the only way that works properly 😄 - https://stackoverflow.com/a/53358384/732221. Also he mentions correctly, that all other overloads of the AddObserver method are broken in Xamarin.